src/Tools/dotnet-monitor/ConfigurationJsonWriter.cs

```diff
@@ -230,12 +232,22 @@ private void ProcessSection(IConfigurationSection section, bool includeChildSect
                     ProcessChildren(section, includeChildSections, redact);
                 }
             }
-            else if (!children.Any())
+            else if (!hasValidChildren)
             {
                 WriteValue(section.Value, redact);
             }
         }
 
+        private bool SectionHasValidChildren(IConfigurationSection section, IEnumerable<IConfigurationSection> children)
```

Comment:
```suggestion
        private bool CanWriteChildren(IConfigurationSection section, IEnumerable<IConfigurationSection> children)
```

Even when the "DiagnosticPort", "DiagnosticPort:ConnectionMode", and "DiagnosticPort:EnpointName" are specified, they are all technically valid (we are not causing configuration to fail nor are the options validation going to fail it). What it's really asking is if it can write the child entries.